src/visualization/components/Header.tsx

```diff
@@ -26,11 +27,36 @@ class Header extends React.Component<any, any> {
         let url  = URL.createObjectURL(blob);
 
         let a = document.createElement('a');
-        a.setAttribute("download", `clarity-${this.props.impression.envelope.impressionId}.json`);
+        a.setAttribute("download", `clarity-${this.props.impression.envelope.impressionId.toUpperCase()}.json`);
         a.href = url;
         a.click();
     }
 
+    uploadJson() {
+        let that = this;
+        let input = document.createElement("input");
+        input.id = "clarityJson";
+        input.type = "file";
+        input.style.display = "none";
+        document.body.appendChild(input);
+        input.onchange = function() {
```

Comment:
>function() { [](start = 25, length = 12)

If you use arrow function here, you'll be able to use 'this' directly without the 'that = this' overhead